src/main/java/org/folio/circulation/domain/validation/LoanAndRelatedRecordsValidator.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.circulation.domain.validation;
+
+import static org.apache.commons.lang3.exception.ExceptionUtils.getStackTrace;
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.results.CommonFailures.failedDueToServerError;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+import java.lang.invoke.MethodHandles;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.domain.LoanRepository;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.ValidationErrorFailure;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoanAndRelatedRecordsValidator {
```

Comment:
How about using any exited validator classes such as ItemStatusValidator or LoanValidator?

Reply:
To maintain separation of concerns in the Object/Class model.
This validator is returning a `Result<LoanAndRelatedRecords>` and not a `Result<Loan>`.
This code is validating the Loan and happens to need the item. It is not quite validating the Item.
These two things tell me that this is more of a `LoanAndRelatedRecordsValidator` than it is either a `LoanValidator` or an `ItemStatusValidator`.

Reply:
`LoanAndRelatedRecords` was an anti-pattern I introduced to help with carry a bag of related things around during asynchronous processes.

Conceptually, a loan and loan with related records are the same.

Some of the validators have overloads for both types.

I believe the loan has an item within it, does the validation require anything other than the loan and the item?